meson.build

```diff
@@ -0,0 +1,246 @@
+    deps += fribidi_dep
+    conf.set('CONFIG_FRIBIDI', 1)
+endif
+
+harfbuzz_options = ['tests=disabled', 'cairo=disabled', 'gobject=disabled']
+if host_machine.system() == 'windows' or host_machine.system() == 'darwin'
+    harfbuzz_options += ['glib=disabled', 'freetype=disabled']
+endif
+harfbuzz_dep = dependency('harfbuzz', version: '>= 1.2.3', required: get_option('harfbuzz'),
+                          fallback: ['harfbuzz', 'libharfbuzz_dep'],
+                          default_options: harfbuzz_options)
+if harfbuzz_dep.found()
+    deps += harfbuzz_dep
+    conf.set('CONFIG_HARFBUZZ', 1)
+endif
+
+req_fontconfig = get_option('fontconfig')
+
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+    req_fontconfig = get_option('fontconfig').disabled()
+endif
+
+fontconfig_dep = dependency('fontconfig', version: '>= 2.10.92', required: req_fontconfig)
+fontconfig = false
+if fontconfig_dep.found()
+    deps += fontconfig_dep
+    conf.set('CONFIG_FONTCONFIG', 1)
+    fontconfig = true
+endif
+
+if get_option('test')
+    deps += dependency('libpng', version: '>= 1.2.0')
+    conf.set('CONFIG_LIBPNG', 1)
+endif
+
+# DirectWrite
+directwrite = false
+if host_machine.system() == 'windows' and not get_option('directwrite').disabled()
+    if cc.has_header('windows.h')
+        deps += cc.find_library('dwrite', required: true)
+        conf.set('CONFIG_DIRECTWRITE', 1)
+        directwrite = true
+    elif get_option('directwrite').enabled()
+        error('DirectWrite was enabled explicitly, but required header is missing.')
+    endif
+endif
+
+# CoreText
+coretext = false
+if host_machine.system() == 'darwin' and not get_option('coretext').disabled()
+    coretext_dep = dependency('appleframeworks', modules: ['CoreText', 'CoreFoundation'], required: false)
+    if cc.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: coretext_dep)
+        deps += coretext_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif get_option('coretext').enabled()
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
+
+if get_option('require-system-font-provider') and not fontconfig and not directwrite and not coretext
+    error('''Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig (Linux, other) is required.
+If you really want to compile without a system font provider, set -Drequire-system-font-provider=false''')
+endif
+
+# dirent.h
+if host_machine.system() == 'windows' and not cc.has_header('dirent.h')
+    deps += subproject('tronkko-dirent').get_variable('tronkko_dirent_dep')
+endif
+
+# ASM
+
+nasm_args = []
+nasm = find_program('nasm', required: get_option('asm'))
+enable_asm = nasm.found()
+
+# NASM version check
+if enable_asm
+    nasm_r = run_command(nasm, '-v')
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
```

Comment:
find_program has a version kwarg which can automatically check this for you by running `nasm --version`, though if it is found and the version is insufficient then it will act as though it was not found and `required: true` will error rather than printing a warning.

Reply:
Yeah, I wanted to match the existing autotools error messages where possible. Maybe it doesn't matter, but when in doubt I erred in that direction.